packages/server/email/renderSSRElement.ts

```diff
@@ -10,7 +10,12 @@ const renderSSRElement = async (element: ReactElement, environment: ServerEnviro
   await environment.load()
 
   // return html string
-  return ReactDOMServer.renderToStaticMarkup(element)
+  const htmlString = ReactDOMServer.renderToStaticMarkup(element)
+
+  // Relay has a memory leak that they won't fix: https://github.com/facebook/relay/pull/2883
+  // To get around it, we destroy everything in the object so while it still exists, it's as small as possible
+  environment.destroy()
```

Comment:
let me dig in! i assumed it was the same cause as it was years ago because the heap dump looked the same. lemme see if i can track down the exact object...

Reply:
They fixed the leak i found! They did not fix the leak that the other dev found: https://github.com/facebook/relay/blob/bc40b4be8eef8ddb0432bd95d55acf3a6d7623de/packages/react-relay/ReactRelayQueryRenderer.js#L54-L59

https://www.loom.com/share/261af0fc96e94b1c8667b6c937379fef

Reply:
it just hit me that `QueryRenderer` is part of the legacy API, perhaps this is fixed if we replace it with `useLazyLoadQuery`. Will try that now...

Reply:
ufff, this is gonna take some work... will revert this PR to draft until complete